server/workflow-client.js

```diff
@@ -293,21 +317,27 @@ WorkflowClient.prototype.openWorkflows = async function({
     typeFilter,
     executionFilter,
   };
-  const res = await this.client.listOpenWorkflowExecutionsAsync(req);
+  const res = await this.client.listOpenWorkflowExecutionsAsync(
+    req,
+    buildGrpcMetadata(metadata)
```

Comment:
suggestion 2: the amount of times you are repeating this line of code suggests that there is some kind of middleware that could be used here

it feels like you are expanding the `openWorkflows` function to receive metadata, only to pass thru metadata to `listOpenWorkflowExecutionsAsync`, and repeating that a few times here. to me a bit of a code smell. i havent looked at this client impl closely but just asking - is there a better way to do what you are trying to do here? are you going to regret adding this in future and have to refactor a bunch?

Reply:
@sw-yx i fully agree i spent some time yesterday thinking about a better approach and didn't finish as it was a bit late, created this PR for visibility. Is this ok if i refactor straight after we test mtls?

Reply:
sgtm

Reply:
followin up for visibility:created an item in internal backlog to track this